HHS/Head-Start-TTADP
# [TTAHUB-1329] Format enddate on backend

## Description of change
This change formats the goal end date on the backend. This comes out of a user reported bug where a goal end date saved on the activity report would immediately disappear on editing it.

## How to test
Confirm that you can save a goal's end date, edit it, leave and return and do the same on the activity report and the RTR create goals page.

To repro existing bug on main:
1) on a new AR, add an existing goal without a date. 
2) Add a date
3) Save draft
4) Leave report (go to landing page)
5) Return to report
6) Date should be gone

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1329


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
